### PR TITLE
Add EfficientNet Presets

### PR DESCRIPTION
**Pull request by pkgoogle**
Ports `efficientnet_b0_ra_imagenet` and `efficientnet_b1_ft_imagenet` presets to EfficientNet and creates the scaffolding for further work

[Tested Colab](https://colab.sandbox.google.com/gist/pkgoogle/678a9b87d4d800c49c700e66d2ffcbc4/densenet-example-keras-hub.ipynb)

### CODE REVIEW

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py [hunk 1]
```diff
@@ -390,25 +408,24 @@ def _apply_efficientnet_block(
...
             kernel_size=kernel_size,
             strides=strides,
-            padding=conv_pad,
+            padding="valid",
+            data_format=data_format,
             use_bias=False,
             depthwise_initializer=conv_kernel_initializer(),
             name=name + "dwconv",
         )(x)
+
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/efficientnet/efficientnet_backbone.py:425]: remove blank line

File: keras_hub/src/utils/timm/convert_efficientnet.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
+import math
```
Comment (divyashreepathihalli) [inline keras_hub/src/utils/timm/convert_efficientnet.py:1]: add instructions on how to run the checkpoint conversion script. see resnet script for example
"""
python ....
"""

**Review by divyashreepathihalli — COMMENTED** [review]
Thanks @pkgoogle. Left a few NIT comments

File: keras_hub/src/models/efficientnet/efficientnet_presets.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+"""EfficientNet preset configurations."""
+
+backbone_presets = {
+    "efficientnet_b0_ra_imagenet": {
+        "metadata": {
+            "description": (
+                "EfficientNet B0 model pre-trained on the ImageNet 1k dataset "
+                "with RandAugment recipe."
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/efficientnet/efficientnet_presets.py:8]: what is the random augment recipie?

Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_presets.py:8]: Here's the [paper](https://arxiv.org/abs/1909.13719)... should we link it?

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py [hunk 4]
```diff
@@ -206,10 +211,13 @@ def __init__(
                         filters_out=output_filters,
                         kernel_size=stackwise_kernel_sizes[i],
                         strides=strides,
+                        data_format=data_format,
                         expand_ratio=stackwise_expansion_ratios[i],
                         se_ratio=squeeze_and_excite_ratio,
                         activation=activation,
+                        project_activation=project_activation,
```
[the remark this answers is not in the document]
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_backbone.py]: done

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py [hunk 5]
```diff
@@ -346,10 +359,13 @@ def _apply_efficientnet_block(
         kernel_size=3,
         strides=1,
         activation="swish",
+        project_activation=None,
```
[the remark this answers is not in the document]
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_backbone.py]: done

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py (hunk 1 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_backbone.py]: done

**Review by divyashreepathihalli — APPROVED** [review]

**Pull request merged**